sklearn/utils/_estimator_html_repr.py

```diff
@@ -472,11 +473,10 @@ def _get_doc_link(self):
         if self._doc_link_url_param_generator is None:
             estimator_name = self.__class__.__name__
             estimator_module = ".".join(
-                [
-                    _
-                    for _ in self.__class__.__module__.split(".")
-                    if not _.startswith("_")
-                ]
+                itertools.takewhile(
```

Comment:
What do you think of adding a comment like "Construct the estimator's module name, up to the first private submodule. This works because in scikit-learn all public estimators are exposed at that level, even if they actually live in a private sub-module"

Basically a short explanation of why this does "the right thing" and that it is somewhat a "lucky accident" because scikit-learn is structure just right.

Reply:
```suggestion
                # Construct the estimator's module name, up to the first private submodule.
                # This works because in scikit-learn all public estimators are exposed at
                # that level, even if they actually live in a private sub-module.
                itertools.takewhile(
```